tests/test_surface_events.cpp

```diff
@@ -534,11 +534,18 @@ TEST_F(ClientSurfaceEventsTest, frame_timestamp_increases)
     usleep(10000);
 
     wl_surface_attach(surface, buffers[2], 0, 0);
+    surface.add_frame_callback(
+        [&](int time)
+        {
+            EXPECT_THAT(time, Gt(prev_frame_time));
+            prev_frame_time = time;
+            frame_callback_count++;
+        });
     wl_surface_commit(surface);
 
     client.dispatch_until([&frame_callback_count]()
         {
-            return frame_callback_count >= 2;
+            return frame_callback_count == 2;
```

Comment:
While we should have a check[*] that `frame_callback_count` is 2, this risks the test hanging if it is more.

[*] Checking this in a test called "frame_timestamp_increases" is wrong: it belongs in "when_two_frames_are_requested_the_two_callbacks_are_seen"

Reply:
I'm pretty sure it's impossible for `frame_callback_count` to exceed 2. We only call `add_frame_callback` twice, and each callback is one-shot. I'm not *against* making it `>=2`, but no compositor behaviour can cause it to exceed 2.

Reply:
:shrug: